pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/client/Auth0ManagementClient.java

```diff
@@ -527,31 +529,55 @@ private ApiResult<String, APIException> tryPasswordChangeRequest(PasswordChangeT
         });
     }
 
-    private <B, E> ApiResult<B, E> withRetries(String retryMessage, Supplier<ApiResult<B, E>> callback) {
+    public <B, E> ApiResult<B, E> withRetries(String retryMessage, Supplier<ApiResult<B, E>> callback) {
         ApiResult<B, E> res = null;
         int numTries = 0;
         int maxTries = maxRetries + 1;
+        long waitTotal = 0L;
         while (numTries < maxTries) {
             res = callback.get();
             numTries++;
             if (numTries >= maxTries) {
                 break;
             }
-            if (res.getStatusCode() == 429) {
+            long wait = backoffMillis * numTries + new Random().nextInt(MAX_JITTER_MILLIS);
+            if (res.getError() instanceof RateLimitException && res.getStatusCode() == 429) {
                 log.error(retryMessage, res.getError());
-                long wait = backoffMillis * numTries + new Random().nextInt(MAX_JITTER_MILLIS);
+                // Get information from auth0 about how long to wait
+                RateLimitException rateLimit = (RateLimitException) res.getError();
+                long retryAfter = auth0BackoffTime(rateLimit);
+                wait = retryAfter > 0 ? retryAfter : wait;
```

Comment:
This test should mirror the value returned by auth0BackoffTime. IOW, it should be `retryAfter != -1` or auth0BackoffTime should return 0 when it does not have a new suggested wait time.